iidm/iidm-api/src/main/java/com/powsybl/iidm/network/Network.java

```diff
@@ -1199,14 +1303,134 @@ default int getConnectableCount() {
+     *
+     * @param subnetworkId the id of a current network's subnetwork.
+     * @return a fully-independent {@link Network} corresponding to the subnetwork.
+     */
+    Network extractSubnetwork(String subnetworkId);
+
+    /**
+     * Check if the given identifiable id <b>directly</b> in the current network.<br/>
+     * If the identifiable is in a subnetwork of the current network, this method will return <code>false</code>.
+     *
+     * @param identifiable the identifiable to check
+     * @return true if the given identifiable is <b>directly</b> in the network
+     */
+    default boolean containsDirectly(Identifiable<?> identifiable) {
+        return identifiable != null && identifiable.getSmallestContainingNetwork() == this;
+    }
+
+    /**
+     * Check if the network contains the given identifiable or another network containing it.
+     *
+     * @param identifiable the identifiable to check
+     * @return true is the given identifiable is in the network (or in one of its subnetworks)
+     */
+    default boolean contains(Identifiable<?> identifiable) {
+        if (containsDirectly(identifiable)) {
+            return true;
+        }
+        return getSubNetworks().stream().anyMatch(s -> s.containsDirectly(identifiable));
+    }
+
+    /**
+     * If both ids correspond to the current network or one of its subnetworks, return a {@link Map} containing
+     * all the equipments linking the both networks (i.e. with a terminal in each network), organized by class.<br/>
+     * Throw an {@link IllegalArgumentException} if one of the given ids doesn't match the known ids of the current network.
+     *
+     * @param networkId1 1st network's id
+     * @param networkId2 2nd network's id
+     * @return the elements linking the 2 given networks, organized by class
+     */
+    default Map<Class<? extends Identifiable<?>>, Set<Identifiable<?>>> getLinksBetween(String networkId1, String networkId2) {
+        return getLinksBetween(getNetworkById(networkId1), getNetworkById(networkId2));
+    }
+
+    /**
+     * If both networks are in the current network (i.e. is the current network or one of its subnetworks), return a {@link Map} containing
+     * all the equipments linking the both networks (i.e. with a terminal in each network), organized by class.
+     *
+     * @param network1 1st network
+     * @param network2 2nd network
+     * @return the elements linking the 2 given networks, organized by class
+     */
+    Map<Class<? extends Identifiable<?>>, Set<Identifiable<?>>> getLinksBetween(Network network1, Network network2);
+
+    /**
+     * Split all links connecting 2 subnetworks of the current network, or between the current network and one of its subnetworks.<br/>
+     * A {@link PowsyblException} is thrown if an un-splittable link is detected.
+     * @param networkId1 1st network's id
+     * @param networkId2 2nd network's id
+     */
+    void splitLinksBetween(String networkId1, String networkId2);
```

Comment:
I am not convinced by all the split methods. The `unmerge` and the `extract` are enough, I don't see the use cases of the split methods you want to expose. Maybe just a feasibility split method to check that the unmerge or the extract are going to be okay or not.

Reply:
yes, the split method is enough

Reply:
I removed the `splitLinksBetween` methods and introduced an other one to check the feasibility and retrieve the equipments preventing the split: `Networks.getSplitPreventingEquipments(Network)`